geode-core/src/main/java/org/apache/geode/internal/cache/BucketAdvisor.java

```diff
@@ -181,6 +182,14 @@ private BucketAdvisor(Bucket bucket, RegionAdvisor regionAdvisor) {
     redundancyTracker =
         new BucketRedundancyTracker(pRegion.getRedundantCopies(), pRegion.getRedundancyTracker());
     resetParentAdvisor(bucket.getId());
+
+    if (parentAdvisor == null) {
```

Comment:
`parentAdvisor` is not `final`, can it be mutated after construction? If so what impact does that have on the inherited locks, which are `final`? If not, can it be made `final`?

Reply:
Make the parentAdvisor to be final now.